gammapy/obs/tests/test_observation.py

```diff
@@ -9,3 +14,181 @@
+    elif type == 'sky_circle':
+        ang_distance = skycoord.separation(center)
+        assert (ang_distance < radius_eff).all()
+
+    # test on the inverted selection
+    selection['inverted'] = True
+    inv_selected_obs_table = obs_table.select_observations(selection)
+    skycoord = skycoord_from_table(inv_selected_obs_table)
+    if type == 'sky_box':
+        skycoord = skycoord.transform_to(selection['frame'])
+        lon = skycoord.data.lon
+        lat = skycoord.data.lat
+        if do_wrapping:
+            lon = lon.wrap_at(Angle(180, 'degree'))
+        assert ((lon_range_eff[0] >= lon) | (lon >= lon_range_eff[1]) |
+                (lat_range_eff[0] >= lat) | (lat >= lat_range_eff[1])).all()
+    elif type == 'sky_circle':
+        ang_distance = skycoord.separation(center)
+        assert (ang_distance >= radius_eff).all()
+
+    # the sum of number of entries in both selections should be the total number of entries
+    assert len(selected_obs_table) + len(inv_selected_obs_table) == len(obs_table)
+
+
+def test_select_parameter_box():
+    # create random observation table
+    observatory_name='HESS'
+    n_obs = 10
+    obs_table = make_test_observation_table(observatory_name, n_obs)
+
+    # test no selection: input and output tables should be the same
+    selected_obs_table = obs_table.select_observations()
+    assert len(selected_obs_table) == len(obs_table)
+
+    # select some pars and check the correspoding values in the columns
+
+    # test box selection in obs_id
+    variable = 'OBS_ID'
+    value_min = 2
+    value_max = 5
+    selection = dict(type='par_box', variable=variable,
+                     value_min=value_min, value_max=value_max)
+    selected_obs_table = obs_table.select_observations(selection)
+    assert (value_min < selected_obs_table[variable]).all()
+    assert (selected_obs_table[variable] < value_max).all()
+
+    # test box selection in obs_id inverted
+    selection = dict(type='par_box', variable=variable,
+                     value_min=value_min, value_max=value_max, inverted=True)
+    selected_obs_table = obs_table.select_observations(selection)
+    assert len(selected_obs_table) == 8
+    assert ((value_min >= selected_obs_table[variable]) |
+            (selected_obs_table[variable] >= value_max)).all()
+
+    # test box selection in alt
+    variable = 'ALT'
+    value_min = Angle(60., 'degree')
+    value_max = Angle(70., 'degree')
+    selection = dict(type='par_box', variable=variable,
+                     value_min=value_min, value_max=value_max)
+    selected_obs_table = obs_table.select_observations(selection)
+    assert (value_min < Angle(selected_obs_table[variable])).all()
+    assert (Angle(selected_obs_table[variable]) < value_max).all()
+
+
+def test_select_time_box():
+    # create random observation table with very close (in time)
+    # observations (and times in absolute times)
+    observatory_name='HESS'
```

Comment:
Same here and again below ... `obseratory_name` should not be specified in this test.